psl-clojure/src/psl_clojure/core.clj

```diff
@@ -0,0 +1,401 @@
+  [data-store]
+  (Model. "" data-store))
+
+(defn model-print
+  "Print the given model."
+  [model]
+  (println (.toString model)))
+
+(defn mpe-inference
+  "Call mpeInference on the inference app."
+  [inference-app]
+  (log/info "inference:: ::starting")
+  (let [result (.mpeInference inference-app)]
+    (log/info "inference:: ::done")
+    result))
+
+(defn open-db
+  "Return an open PSL database."
+  ([datastore model parts-to-read part-to-write preds-to-close]
+     {:pre [(not-any? nil? [datastore model parts-to-read part-to-write preds-to-close])]}
+     (let [parts-to-read (into-array Partition parts-to-read)
+           preds-to-close (HashSet.
+                           (for [pname preds-to-close]
+                             (.getPredicate model pname)))]
+       (.getDatabase datastore part-to-write preds-to-close parts-to-read)))
+
+  ;; No predicates to close
+  ([datastore model parts-to-read part-to-write]
+     {:pre [(not-any? nil? [datastore model parts-to-read part-to-write])]}
+     (let [parts-to-read (into-array Partition parts-to-read)]
+       (.getDatabase datastore part-to-write parts-to-read)))
+
+  ;; No predicates to close or write partition
+  ([datastore model parts-to-read]
+     {:pre [(not-any? nil? [datastore model parts-to-read])]}
+     (let [parts-to-read (into-array Partition parts-to-read)]
+       (.getDatabase datastore (first parts-to-read) parts-to-read))))
+
+(defn p
+  "Get a predicate object from the PSL model"
+  [model pred-name]
+  (.getPredicate model pred-name))
+
+;;; =============== Functions for handling PSL partitions ====================
+
+(defn partition-copy-atoms
+  "Copy atoms of given predicates from one partition to another"
+  [datastore model part-from part-to preds]
+  (let [dbr (open-db datastore model [part-from])
+        dbw (open-db datastore model [part-to] part-to)]
+    (try
+      (doseq [pnam preds]
+        (let [atoms (Queries/getAllAtoms dbr (p model pnam))]
+          (doseq [atom atoms]
+            (.commit dbw atom))))
+      nil
+      (finally (close-db dbr)
+               (close-db dbw)))))
+
+(defn partition-delete
+  "Delete the contents of the partition in the datastore."
+  [datastore partition] (.deletePartition datastore partition))
+
+(defn partitions-delete
+  "Delete the contents of the partitions in the datastore."
+
+  ;; Selected partitions
+  ([datastore partitions]
+   (doseq [partition partitions]
+     (partition-delete datastore partition))
+   (count partitions))
+
+  ;; All partitions
+  ([datastore]
+   (let [partitions (.listPartitions datastore)]
+     (doseq [partition partitions]
+       (partition-delete datastore partition))
+     (count partitions))))
+
+(defn partition-new
+  "Return a new partition within this session."
+  [datastore]
+  (.getNewPartition datastore))
+
+;;; =============== Functions for handling data in PSL ====================
+
+(defn pred-append
```

Comment:
Previous conversation:
```
eriq-augustine 7 days ago  Owner
I would also pass through some of the Inserter file loading methods.

alexmemory 4 days ago
Agreed. These data handling functions are geared towards making Incanter (like pandas or R, but in JVM) seamlessly interoperable with PSL. So, to load from files you'd use the built-in capabilities of Incanter. But I'm guessing there are advantages of Inserter, such as bulk load?
```

Yes, the file-based loading methods will bulk load whenever possible.
Since this interface is currently pretty thick (as opposed to a thin one), you should also be aware of more database semantics.
Database tables are not indexed on predicate creation.
They are only indexed on the first call to Datastore.getDatabase().
This is so that people can insert into the data store (tables) without having an index on top of if.
Make sure that non of these Clojure workflows violate that.
It will still work, but be slower.